homeassistant/components/holiday/config_flow.py

```diff
@@ -47,7 +47,7 @@ async def async_step_user(
             self._async_abort_entries_match({CONF_COUNTRY: user_input[CONF_COUNTRY]})
 
             try:
-                locale = Locale(self.hass.config.language.replace("-", "_"))
+                locale = Locale.parse(self.hass.config.language, sep="-")
```

Comment:
Works fine if there is no separator too?

Reply:
Yes, locally tested, and otherwise https://github.com/home-assistant/core/blob/dev/tests/components/holiday/test_config_flow.py#L73 should fail, right?